src/disk.c

```diff
@@ -63,6 +63,140 @@ static PedExceptionOption disk_exception_handler(PedException* ex) {
 	return PED_EXCEPTION_UNHANDLED;
 }
 
+static char *blk_device_by_path(const gchar* path) {
+	struct stat st = { 0 };
+
+	if (!path) {
+		LOG(MOD "Path is empty\n");
+		return NULL;
+	}
+	if (stat(path, &st) != 0) {
+		LOG(MOD "Cannot stat '%s'\n", path);
+		return NULL;
+	}
+
+	return blkid_devno_to_devname(st.st_dev);
+}
+
+static PedPartition *get_rootfs_partition(PedDisk* disk) {
+	char *rootfs_dev_path = NULL;
+	char *part_path = NULL;
+	PedPartition *part = NULL;
+	int last_partition_num = 0;
+	int i = 0;
+
+	rootfs_dev_path = blk_device_by_path("/");
+
+	if (!rootfs_dev_path) {
+		LOG(MOD "Could not get the rootfs block device\n");
+		return NULL;
+	}
+
+	/* Looking for rootfs partition */
+	last_partition_num = ped_disk_get_last_partition_num(disk);
+	for(i = 1; i <= last_partition_num; i++) {
+		part = ped_disk_get_partition(disk, i);
+		part_path = ped_partition_get_path(part);
+		if (!part_path) {
+			LOG(MOD "Could not get partition path. Start sector: %lld\n",
+			    part->geom.start);
+			continue;
+		}
+
+		if (strncmp(part_path, rootfs_dev_path, strlen(rootfs_dev_path)) == 0) {
+			LOG(MOD "Found rootfs in block device: %s. Start sector: %lld\n",
+			    rootfs_dev_path, part->geom.start);
+			free(part_path);
+			goto out;
+		}
+
+		free(part_path);
+	}
+
+out:
+	free(rootfs_dev_path);
+	return part;
+}
+
+static gboolean resize_rootfs_blk_device(PedDisk* disk) {
+	PedPartition *rootfs_part = NULL;
+	PedGeometry geometry_start;
+	PedGeometry* geometry_end = NULL;
+	PedConstraint* constraint = NULL;
+	PedPartition* nextPartition = NULL;
```

Comment:
All of these may need `free()`ing after error/exit?

Reply:
yes, here https://github.com/clearlinux/micro-config-drive/pull/35/files#diff-352f7d19b14cf77c9be0554e611634faR192 ,  structures returned by `ped_disk_get_partition()` are not freed as we can see in this test https://fossies.org/linux/parted/libparted/tests/disk.c